### PR TITLE
Add sigs validator

### PR DESCRIPTION
**Pull request by dhiller**
This PR adds a tool that can validate the basics of the `sigs.yaml` file. Currently it can validate/remove non existing links to owners elements.

Also it updates the sigs.yaml with removed unreachable owners links.

/cc @mazzystr @fabiand @fgimenez

### CODE REVIEW

**Comment by dhiller** [comment]
@mazzystr sorry, had to fix error regarding the devstats generator, PTAL! Thanks!

**Comment by dhiller** [comment]
Please see kubevirt/project-infra#1160, there I have added a validation presubmit, so that we can be sure it is correct whenever it changes.

**Comment by mazzystr** [comment]
/approve

@dhiller do you want anyone else to approve?

File: validators/cmd/sigs/sigs-validator.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+type options struct {
+	dryRun       bool
+	sigsFilePath string
+}
+
+func (o *options) Validate() error {
+	if o.sigsFilePath == "" {
+		return fmt.Errorf("path to sigs.yaml is required")
+	} else {
```
Comment (fgimenez) [inline validators/cmd/sigs/sigs-validator.go:76]: I think we can remove tihs `else`

File: validators/cmd/sigs/sigs-validator.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+				response, err := http.DefaultClient.Head(ownersFileURL)
+				if err != nil {
+					log.Printf("failed to retrieve %s, continuing with next", ownersFileURL)
+					continue
+				}
+				defer response.Body.Close()
+				if response.StatusCode >= http.StatusOK && response.StatusCode < http.StatusMultipleChoices {
+					foundOwners = append(foundOwners, ownersFileURL)
+				} else {
+					log.Printf("failed to retrieve %s", ownersFileURL)
```
Comment (fgimenez) [inline validators/cmd/sigs/sigs-validator.go:128]: Would be good to add `response.StatusCode` here.

**Review by fgimenez — COMMENTED** [review]
Looks good, just a couple of suggestions.

**Comment by dhiller** [comment]
@fgimenez addressed review comments, PTAL! Thanks!

**Review by fgimenez — APPROVED** [review]
/lgtm

Thanks!

**Pull request merged**